README.md

```diff
@@ -260,19 +260,60 @@ While using the repository from SBT, you can use any version of Apache Spark you
 * you run the sbt command using the wanted version `sbt -Dspark.version=1.4.0-SNAPSHOT`
 * there is a folder in `modules/spark/src/main/scala-2.1X` that points to the base of the required version (excl. the classifier SNAPHOST or RC): `spark-1.4`
 
+#### Building for specific distributions
+
+##### MapR
+
+##### Building for MapR
+MapR has a custom set of Hadoop jars that must be used.  To build using these jars:
+* Add the [MapR Maven repository](http://doc.mapr.com/display/MapR/Maven+Artifacts+for+MapR) as an sbt
+[proxy repository](http://www.scala-sbt.org/0.13/docs/Proxy-Repositories.html)
+* Build Spark Notebook using the MapR Hadoop jars by setting the `hadoop-version` property to the MapR-specific version.
+See the MapR Maven Repository link above for the specific versions to use.  For example, to build Spark Notebook with
+the MapR Hadoop jars and Hive and Parquet support for Spark 1.3.1:
+```
+sbt -Dspark.version=1.3.1 -Dhadoop.version=2.5.1-mapr-1503 -Dwith.hive=true -Dwith.parquet=true clean dist
+```
+
+##### Running with MapR
+A few extra jars need to be added to the app classpath:
+* `commons-configuration`
+* `hadoop-auth`
+* `maprfs`
+
+Then add them to the classpath in the `spark-notebook` script.  For example:
+```
+declare -r app_classpath="${HADOOP_CONF_DIR}:<long list of jars>:
```

Comment:
I was actually thinking exactly the same an hour ago. So we would duplicate the hadoop one everywhere it's used

* https://github.com/andypetrella/spark-notebook/blob/master/build.sbt#L92
* https://github.com/andypetrella/spark-notebook/blob/master/modules/subprocess/src/main/scala/notebook/kernel/pfork/ProcessFork.scala#L84
* https://github.com/andypetrella/spark-notebook/blob/master/modules/subprocess/src/main/scala/notebook/kernel/pfork/BetterFork.scala#L115

I was also thinking at a custom parameter that would also allow the user to specify extra var env in fact. I'll create an entry for that.
And thus the HADOOP_CONF_DIR, EXTRA_CLASSPATH will still be taken into account, but also others if needed.

Will create issues for both the extra thingy and the custom.

Meanwhile, If you're okay, you can directly add the EXTRA_CLASSPATH support (since the docs will rely on it) in this PR, and mention that it will fix both entries (mapr and extra)

wdyt?